src/charm.py

```diff
@@ -124,6 +124,17 @@ def __init__(self, *args):
         patroni_api_port = ServicePort(8008, name="api")
         self.service_patcher = KubernetesServicePatch(self, [postgresql_db_port, patroni_api_port])
 
+    def _generate_metrics_jobs(self, enable_tls: bool) -> Dict:
+        """Generate spec for Prometheus scraping."""
+        return [
+            {"static_configs": [{"targets": [f"*:{METRICS_PORT}"]}]},
+            {
+                "static_configs": [{"targets": ["*:8008"]}],
+                "scheme": "https" if enable_tls else "http",
+                "tls_config": {"insecure_skip_verify": True},
```

Comment:
Certificate is likely self signed

Reply:
IMHO, we must use insecure_skip_verify for self-signed certificates only.
If CA is real -> real cert with CA known to grafana-agent => we should not set this option.
Please report it as a TODO ticket in backlog (attached to the topic of will TLS support on DB<>COS, as we need to add TLS support for *_exporters too).

Reply:
[DPE-2133](https://warthogs.atlassian.net/browse/DPE-2133)